crates/contracts/solidity/Makefile

```diff
@@ -0,0 +1,46 @@
+DOCKER := docker
+JQ     := jq
+
+SOLC     := ethereum/solc:0.8.16
```

Comment:
> Just overall consistency?

The main motivations are:
- No unexpected failures because of breaking compiler changes if we keep to the latest
- We need at least 0.8.15 because of the new `encodeCall` built-in
- compiler output changes across versions; we would introduce non-reproducible `Contract.json` artifacts into the repo, as it would depend on the compiler version you are using.